e2e/utils/simulator/CreateSimulationDriver.ts

```diff
@@ -0,0 +1,205 @@
+import { expect, Page } from '@playwright/test';
+import {
+  assertDebugToken,
+  assertOverlappingTestCase,
+  assertRecurringTestCase,
+  getRecurringSettings,
+  screenshotPath,
+} from './utils';
+import {
+  RangeOrder,
+  debugTokens,
+  CreateStrategyTestCase,
+  Direction,
+  Setting,
+  StrategyType,
+} from './types';
+import { waitModalClose, waitModalOpen, waitTooltipsClose } from '../modal';
+import { screenshot, shouldTakeScreenshot, waitFor } from '../operators';
+import { MainMenuDriver } from '../MainMenuDriver';
+import { dayjs } from '../../../src/libs/dayjs';
+
+export class CreateSimulationDriver {
+  constructor(private page: Page, private testCase: CreateStrategyTestCase) {}
+
+  async waitForPriceChart(timeout?: number) {
+    await waitFor(this.page, 'price-chart', timeout);
+  }
+
+  async screenshotPriceChart() {
+    const priceChart = this.page.getByTestId('price-chart');
+    await priceChart.scrollIntoViewIfNeeded();
+    await screenshot(
+      priceChart,
+      screenshotPath(this.testCase, 'simulator-input-price')
+    );
+  }
+
+  async waitForDisclaimerModal(timeout?: number) {
+    await waitFor(this.page, 'sim-disclaimer-modal', timeout);
+  }
+
+  getForm() {
+    return this.page.getByTestId('create-simulation-form');
+  }
+
+  getFormSection(direction: Direction) {
+    const form = this.page.getByTestId(`${direction}-section`);
+    return {
+      locator: form,
+      price: () => form.getByTestId('input-price'),
+      min: () => form.getByTestId('input-min'),
+      max: () => form.getByTestId('input-max'),
+      budget: () => form.getByTestId('input-budget'),
+      setting: (setting: 'limit' | 'range') => {
+        return form.getByTestId(`tab-${setting}`);
+      },
+    };
+  }
+
+  async selectMonthYear(monthYear: string) {
+    const findMonthYear = async () => {
+      const isMonthYearVisible = await this.page.isVisible(
+        `text='${monthYear}'`
+      );
+      if (isMonthYearVisible) {
+        return;
+      } else {
+        await this.page.getByTestId('date-picker-left-arrow').click();
+        return await findMonthYear();
+      }
+    };
+    await findMonthYear();
+  }
```

Comment:
small display feedback.
- Since prettier makes the code uglier by capping the amount of character I tend to split the code into more variables
- When returning earlier I tend to avoid `else` statements
- Almost never use `return await`. Since `findMonthYear` is already a promise you'll return it and the caller will await
```suggestion
  selectMonthYear(monthYear: string) {
    const findMonthYear = async () => {
      const selector = `text='${monthYear}'`;
      const isMonthYearVisible = await this.page.isVisible(selector);
      if (isMonthYearVisible) return;
      await this.page.getByTestId('date-picker-left-arrow').click();
      return findMonthYear();
    };
    return findMonthYear();
  }
```